internal/pkg/cli/flag.go

```diff
@@ -16,21 +16,22 @@ import (
 // Long flag names.
 const (
 	// Common flags.
-	nameFlag         = "name"
-	appFlag          = "app"
-	envFlag          = "env"
-	workloadFlag     = "workload"
-	svcTypeFlag      = "svc-type"
-	jobTypeFlag      = "job-type"
-	typeFlag         = "type"
-	profileFlag      = "profile"
-	yesFlag          = "yes"
-	jsonFlag         = "json"
-	allFlag          = "all"
-	forceFlag        = "force"
-	noRollbackFlag   = "no-rollback"
-	manifestFlag     = "manifest"
-	resourceTagsFlag = "resource-tags"
+	nameFlag           = "name"
+	appFlag            = "app"
+	envFlag            = "env"
+	workloadFlag       = "workload"
+	svcTypeFlag        = "svc-type"
+	jobTypeFlag        = "job-type"
+	typeFlag           = "type"
+	profileFlag        = "profile"
+	yesFlag            = "yes"
+	jsonFlag           = "json"
+	allFlag            = "all"
+	forceFlag          = "force"
+	allowDowngradeFlag = "allow-downgrade"
```

Comment:
We have the `--yes` and `--diff-yes` flags. Perhaps we can consider `--downgrade-yes` just to be consistent!

Reply:
Both `--yes` and `--diff-yes` are for skipping conformation question to unblock CICD or script users. I don't think it is the same case here 🤔

Reply:
Just thinking out loud here to be clear on the usage of this flag 💭 

```
# The app template is different in v1.30.0 vs. v1.28.0.
$ copiot@v1.30.0 app init
$ copilot@v1.28.0 env init  # Error out!

# Or
$ copiot@v1.26.0 app init
$ copilot@v1.30.0 app upgrade
$ copilot@v1.28.0 env init  # Error out!
```

People could just run `$ copilot@v1.30.0 env init` instead, so that it wouldn't trigger a downgrade.

Then why would a user want to force an app downgrade? There are several possibilities:
1. They _have to_ run `env init` with `v1.28.0` (due to bugs, breaking changes, etc.).
2. They want to downgrade the app template, but there is no `app deploy` nor `app downgrade`. They have to use `env init` to do so.

Reply:
What do you think of `--force-app-downgrade` then?

Reply:
> They want to downgrade the app template, but there is no app deploy nor app downgrade. They have to use env init to do so.

I was on the fence about this (actually i reverted my changes for `app upgrade` for adding this flag). The main reason is because using `app upgrade` sounds weird. Maybe it's not that strange if they only need to downgrade the app template?

> What do you think of --force-app-downgrade then?

`--force-app-downgrade` sounds to me that we'll `force` anyway. Like `git push -f` creates a new commit id no matter there's any change or not. For our downgrade flag it's like only downgrade if needed

```
$ copiot@v1.30.0 app init
$ copilot@v1.30.0 env init  --force-app-downgrade
```
won't downgrade anyway...

Reply:
> I was on the fence about this (actually i reverted my changes for app upgrade for adding this flag). The main reason is because using app upgrade sounds weird. Maybe it's not that strange if they only need to downgrade the app template?

Ah it's fine - I'm not proposing that we support `app downgrade` command, or support downgrading from `app upgrade` (also don't think we should, at least not now). I'm just trying to think why users will want to run `env init --downgrade`.

> --force-app-downgrade sounds to me that we'll force anyway.

That makes sense to me. This reminds me of `--force` in service. Both flags perform the action optionally even if the flag is there. 

I guess `--allow` makes the most sense after all!